P/pybind11/build_tarballs.jl

```diff
@@ -0,0 +1,42 @@
+using BinaryBuilder
+
+name = "pybind11"
+version = v"2.5.0"
+
+sources = [
+    ArchiveSource(
+        "https://github.com/pybind/pybind11/archive/v$(version).tar.gz",
+        "97504db65640570f32d3fdf701c25a340c8643037c3b69aec469c10c93dc8504"
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/pybind11-*
+mkdir build
+cd build
+
+cmake .. \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DPYTHON_EXECUTABLE="${bindir}/python3.8${exeext}" \
+    -DPYBIND11_TEST=OFF
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [AnyPlatform()]
```

Comment:
@pabloferz as far as I can tell this builder produces only header files, and from what I can see they're platform-independent (at least between the 3 platforms I was able to check it).  In this case we can build for [`AnyPlatform`](https://juliapackaging.github.io/BinaryBuilder.jl/dev/reference/#BinaryBuilder.AnyPlatform).  If you're happy with these change we can merge it